### ISSUE TITLE
ComboBox: TypeError: Cannot read properties of null (reading 'length')

### ISSUE THREAD

**Issue by vigrase** [SAP/ui5-webcomponents#8634]
### Bug Description

In file ComboBox.js, when value is not specified and you click on it it throws error

![image](https://github.com/SAP/ui5-webcomponents/assets/155465343/ad44f126-e4b4-42a1-9b3a-c87474e64e38)

## Browser
Chrome

### Affected Component

ComboBox

### Expected Behaviour

If the value is undefined or null, behave as the value undefined is allowed and not throw error.

Value undefined is allowed by definition:
![image](https://github.com/SAP/ui5-webcomponents/assets/155465343/45143146-0410-4906-84b7-4e90c2a810f8)

### Steps to Reproduce

1. Create a ComboBox
2. Add several items
3. Pass to ComboBox undefined as value
4. Click on it

### Priority

Medium

### UI5 Web Components Version

1.21.0

### Browser

Chrome, Edge

### Operating System

Windows, MacOS

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by dobrinyonkov** [comment]
Hi @vigrase, Thank you for reporting this issue.

It seems there might be confusion regarding TypeScript's handling of optional properties.

In TypeScript, when a property is declared optional using the ? syntax in an interface, it means that the property can either have a defined value of a certain type or be absent altogether. This doesn't imply that the property can be explicitly set to null or undefined unless explicitly stated.

**To clarify, the value property of the combobox component is defined as optional with the type string (value?: string). This indicates that the property can be set with a string value or be absent/omitted.** 

As shown in the documentation the value property should be a string - https://sap.github.io/ui5-webcomponents/playground/?path=/docs/main-combo-box--combo-box-overview
![image](https://github.com/SAP/ui5-webcomponents/assets/22766569/e8ba42a2-4e5e-43ac-ab5a-107ad9f4b23c)



However, setting it explicitly to null might lead to runtime errors because the component might not handle null values correctly. To ensure proper handling of optional properties, **TypeScript offers a feature we suggest enabling called "Exact Optional Property Types".** See https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes .
When enabled, TypeScript enforces stricter rules regarding optional properties, ensuring that they can only be assigned values that are explicitly allowed by the type definition.

For instance, consider the following interface:
```typescript
interface UserDefaults { 
    colorThemeOverride?: "dark" | "light"; 
}
```

Without "Exact Optional Property Types" enabled, TypeScript allows setting colorThemeOverride to "dark", "light", or undefined. However, enabling "Exact Optional Property Types" ensures that only these values are permitted, disallowing null or other unexpected values.

In summary, to address the reported issue with the combobox component, it's essential to ensure that the value property is handled according to its type definition (for example, by setting empty string, instead of null), and if necessary, enable "Exact Optional Property Types" in TypeScript to enforce stricter rules regarding optional properties.

Regards,
Dobrin
